app/assets/javascripts/miq_qe.js

```diff
@@ -0,0 +1,86 @@
+ManageIQ.qe.get_debounce_index = function () {
+  if (ManageIQ.qe.debounce_counter > 30000) {
+    ManageIQ.qe.debounce_counter = 0;
+  }
+  return ManageIQ.qe.debounce_counter++;
+}
+
+if (typeof _ !== 'undefined' && typeof _.debounce !== 'undefined') {
+  var orig_debounce = _.debounce;
+  _.debounce = function(func, wait, options) {
+    var debounce_index = ManageIQ.qe.get_debounce_index();
+
+    // Override the original fn; new_func will be the original fn with wait prepended to it
+    // We make sure that once this fn is actually run, it decreases the counter
+    var new_func = function() {
+      try {
+        return func.apply({}, arguments);
+      } finally {
+        // this is run before the return above, always
+        delete ManageIQ.qe.debounced[debounce_index];
+      }
+    }
+    // Override the newly-created fn (prepended wait + original fn)
+    // We have to increase the counter before the waiting is initiated
+    var debounced_func = orig_debounce.call(this, new_func, wait, options);
+    var new_debounced_func = function() {
+      ManageIQ.qe.debounced[debounce_index] = 1;
+      return debounced_func.apply(this, arguments);
+    }
+    return new_debounced_func;
+  }
+}
+
+ManageIQ.qe.xpath = function(root, xpath) {
+  if (root == null) {
+     root = document;
+  }
+  return document.evaluate(xpath, root, null,
+    XPathResult.ANY_UNORDERED_NODE_TYPE, null).singleNodeValue;
```

Comment:
That's just a function that the QE guys wanted in. And I doubt they call it from IE.